src/main/scala/com/fulcrumgenomics/commons/util/SystemUtil.scala

```diff
@@ -0,0 +1,79 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.commons.util
+import scala.util.matching.Regex
+
+
+/** System utility defaults and methods. Many methods mock functions in `org.apache.commons:commons-lang3`. */
+object SystemUtil {
+
+  /** The operating system name prefixes for Linux */
+  private val LinuxNamePrefixes: Seq[String] = Seq("Linux", "LINUX")
+
+  /** The operating system name prefixes for Mac */
+  private val MacNamePrefixes: Seq[String] = Seq("Mac")
+
+  /** The current operating system name. */
+  lazy val OsName: Option[String] = getSystemProperty(property = "os.name")
+
+  /** The current operating system architecture */
+  lazy val OsArch: Option[String] = getSystemProperty(property = "os.arch")
+
+  /** The current operating system version */
+  lazy val OsVersion: Option[String] = getSystemProperty(property = "os.version")
+
+  /** Gets a system property.  Returns None if not found or we are not allowed to retrieve the property. */
+  def getSystemProperty(property: String): Option[String] = {
+    try { Option(System.getProperty(property)) }
+    catch { case _: SecurityException => None } // not allowed to look at this property
+  }
+
+  /** True if this operating system is Linux, false otherwise. */
+  lazy val IsOsLinux: Boolean = LinuxNamePrefixes.exists(prefix => OsName.exists(_.startsWith(prefix)))
+
+  /** True if this operating system is Mac, false otherwise. */
+  lazy val IsOsMac: Boolean = MacNamePrefixes.exists(prefix => OsName.exists(_.startsWith(prefix)))
+
+  /** Returns true if the architecture is the given name, false otherwise. */
+  def isOsArch(name: String): Boolean = OsArch.contains(name)
+
+  /** Returns true if the architecture matches the given regular expression, false otherwise. */
+  def isOsArch(regex: Regex): Boolean = OsArch.exists(v => regex.findFirstMatchIn(v).isDefined)
+
+  /** Returns true if the operating system version is the given name, false otherwise. */
+  def isOsVersion(name: String): Boolean = OsVersion.contains(name)
+
+  /** Returns true if the operating system version matches the given regular expression, false otherwise. */
+  def isOsVersion(regex: Regex): Boolean = OsVersion.exists(v => regex.findFirstMatchIn(v).isDefined)
```

Comment:
The member `Regex.matches()` doesn't exist in Scala 2.12 so I moved these functions to `findFirstMatchIn()`.

But I that's not correct because we want full match anchoring.

For example. This is now possible, but shouldn't be:

```scala
scala> OsVersion
Some(0...0.00.0..0.10.14.6)
scala> isOsVersion("10\\.14\\.".r)
true
```

I'll try again.